lib/WWW/WebKit2.pm

```diff
@@ -383,12 +383,14 @@ sub pending {
 sub process_events {
     my ($self) = @_;
 
+    Gtk3::main_iteration_do(0);
     Gtk3::main_iteration while Gtk3::events_pending;
```

Comment:
In a multi threaded program, this loop smells like a TOCTOU [1] race condition.
`Gtk3::main_iteration` processes one event and blocks if no event is available. That's why we check if there even are events to be processed. But what if an event can also disappear from the queue between our check of `Gtk3::events_pending` and the call to `Gtk3::main_iteration`? I don't know if that's possible, but in any case it seems safer to always use `Gtk3::main_iteration_do(0)` instead, since that will not block if the event queue is empty.

[1] https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use